internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -112,10 +116,27 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 	// NOTE: Once we reconcile node labels for the first time, the NodeUninitializedTaint is removed from the node.
 	nodeLabels := getManagedLabels(machine.Labels)
 
+	// Get interruptible instance status from the infrastructure provider and set the interruptible label on the node.
+	interruptible := false
+	if infra != nil {
+		interruptible, _, err = unstructured.NestedBool(infra.Object, "status", "interruptible")
+		if err != nil {
+			log.V(1).Error(err, "Failed to get interruptible status from infrastructure provider")
+		}
+		// If interruptible is true add the interruptible label to the node labels.
+		if interruptible {
+			nodeLabels[clusterv1.InterruptibleLabel] = ""
+		}
+	}
```

Comment:
```suggestion
	interruptible := false
	found := false
	if infra != nil {
		interruptible, found, err = unstructured.NestedBool(infra.Object, "status", "interruptible")
		if err != nil {
			return ctrl.Result{}, errors.Wrapf(err, "failed to get status interruptible from infra machine %s", klog.KObj(infra))
		}
		// If interruptible is true add the interruptible label to the node labels.
		if found && interruptible {
			nodeLabels[clusterv1.InterruptibleLabel] = ""
		}
	}
```
Let's improve this while we're here. Just logging an error if the field exists but is not a bool is an awkward way to handle this

(I know the old implementation was the same)